fix: remove ts expect error

IWANABETHATGUY opened the pull request:
<!-- Thank you for contributing! -->

### Description
Closed #1811 
<!-- Please insert your description here and provide especially info about the "what" this PR is solving -->

IWANABETHATGUY commented on packages/rolldown/src/utils/normalize-hook.ts:
```diff
@@ -1,25 +1,20 @@
+	: never {
+	if (typeof hook === "function") {
+    // @ts-ignore
+		return [hook, {}];
```
previously: 
![image](https://github.com/user-attachments/assets/a7445366-5ea9-491d-a02a-765d042213f4)
current: 
![image](https://github.com/user-attachments/assets/01fdbd25-f147-4336-aa9e-e01cf651bab5)

IWANABETHATGUY replied:
Sadly, it still needs ts **directive**, forgive me poor **type gymnastics**

underfin commented on packages/rolldown/src/utils/normalize-hook.ts:
```diff
@@ -1,25 +1,19 @@
+  hook: T,
+): T extends ObjectHook<infer A, infer B>
+  ? T extends Function
+    ? [A, {}]
```
I refactor the function to make support `ObjectHook<string, {}>` for AddonHooks, and remove using `infer` make typing clear, the `@ts-expect-error` is not used.

IWANABETHATGUY replied:
I see the pr, but it lost the ability to infer the type

IWANABETHATGUY replied:
And it may cause more type errors in the down stream 
![image](https://github.com/user-attachments/assets/c56b7240-50c6-4863-9489-33d0007644ff)

IWANABETHATGUY commented:
I am gonna merge this, but I am open to accept if any one have a better idea

IWANABETHATGUY committed:
fix: remove ts expect error (#1817)

<!-- Thank you for contributing! -->

### Description
Closed #1811
<!-- Please insert your description here and provide especially info about the "what" this PR is solving -->
